diff --git a/p_0/backend/Utils/object_utils.py b/p_0/backend/Utils/object_utils.py
index d8ba9d4..bcb20aa 100644
--- a/p_0/backend/Utils/object_utils.py
+++ b/p_0/backend/Utils/object_utils.py
@@ -65,7 +65,7 @@ def prepare_download(bucketName,objectName,startb,endb,mongo):
 def metadata_adder(bucketName,objectName,key,request,mongo):
     obj = validate_object(bucketName,objectName,mongo)
     if obj:
-        obj["metadata"][key] = request.data
+        obj["metadata"][key] = request.data.decode("utf8")
         mongo.db[bucketName].save(obj)
         return True
     else:
diff --git a/p_2/worker/Dockerfile b/p_2/worker/Dockerfile
index c4fd483..57635c1 100644
--- a/p_2/worker/Dockerfile
+++ b/p_2/worker/Dockerfile
@@ -14,8 +14,10 @@ WORKDIR /app
 
 
 RUN mkdir -p /app/out
-RUN apk --update add --no-cache file imagemagick build-base jpeg-dev zlib-dev && \
-apk add fontconfig ttf-Impact
+RUN apk --no-cache add msttcorefonts-installer fontconfig && \
+    update-ms-fonts && \
+    fc-cache -f
+RUN apk --update add --no-cache file imagemagick build-base jpeg-dev zlib-dev
 ENV MAGICK_HOME=/usr
 RUN apk add --no-cache imagemagick-dev
 
